src/NLog/Targets/WebServiceTarget.cs

```diff
@@ -312,135 +335,302 @@ private Uri BuildWebServiceUrl(object[] parameterValues)
         }
 
-        private MemoryStream PrepareSoap12Request(HttpWebRequest request, object[] parameterValues)
+        private class HttpPostFormEncodedFormatter : HttpPostTextFormatterBase
         {
-            return PrepareSoapRequestPost(request, parameterValues, Soap12EnvelopeNamespace, "soap12");
+            UrlHelper.EscapeEncodingFlag encodingFlags;
+
+            public HttpPostFormEncodedFormatter(WebServiceTarget target) : base(target)
+            {
+                encodingFlags = UrlHelper.GetUriStringEncodingFlags(target.EscapeDataNLogLegacy, true, target.EscapeDataRfc3986);
+            }
+
+            protected override string ContentType
+            {
+                get { return "application/x-www-form-urlencoded"; }
+            }
+
+            protected override string Separator
+            {
+                get { return "&"; }
+            }
+
+            protected override string GetFormattedContent(string parametersContent)
+            {
+                return parametersContent;
+            }
+
+            protected override string GetFormattedParameter(MethodCallParameter parameter, object value)
+            {
+                string parameterValue = Convert.ToString(value, CultureInfo.InvariantCulture);
+                string valueString = string.Empty;
+                if (!string.IsNullOrEmpty(parameterValue))
+                {
+                    var sb = new StringBuilder(Math.Max(parameterValue.Length + 20, Target.Parameters.Count > 1 ? 256 : 0));
+                    UrlHelper.EscapeDataEncode(parameterValue, sb, encodingFlags);
+                    valueString = sb.ToString();
+                }
+
+                return string.Format("{0}={1}",
+                        parameter.Name,
+                        valueString);
+            }
         }
 
-        /// <summary>
-        /// Helper for creating soap POST-XML request
-        /// </summary>
-        /// <param name="request"></param>
-        /// <param name="parameterValues"></param>
-        /// <param name="soapEnvelopeNamespace"></param>
-        /// <param name="soapname"></param>
-        /// <returns></returns>
-        private MemoryStream PrepareSoapRequestPost(WebRequest request, object[] parameterValues, string soapEnvelopeNamespace, string soapname)
+        private class HttpPostJsonFormatter : HttpPostTextFormatterBase
+        {
+            public HttpPostJsonFormatter(WebServiceTarget target) : base(target)
+            { }
+
+            protected override string ContentType
+            {
+                get { return "application/json"; }
+            }
+
+            protected override string Separator
+            {
+                get { return ","; }
+            }
+
+            protected override string GetFormattedContent(string parametersContent)
+            {
+                return string.Format("{{{0}}}", parametersContent);
```

Comment:
Consider changing into this:

return string.Concat("{{", parametersContent, "}}");